Some score tests fail

```
$ py.test
rose/server/player_test.py ..
rose/server/score_test.py FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF
```

Last commit where the tests passed is 73afb536d81c86336cd37d6890111d1611f4ce91

Looks like 49159b40aa1a1ea01b30fdc4d8a0dc016b5ed9a9 broke the tests.

Some tests were fixed in #152, we still have these failures:

```
==================================================================================== test session starts =====================================================================================
platform linux2 -- Python 2.7.13, pytest-3.1.3, py-1.4.33, pluggy-0.4.0
rootdir: /home/nsoffer/src/rose, inifile:
plugins: timeout-1.2.0, threadleak-0.1.0, cov-2.4.0
collected 34 items 

rose/server/player_test.py ..
rose/server/score_test.py ..F...F.F...F....F..F..F........

========================================================================================== FAILURES ==========================================================================================
________________________________________________________________________________ TestNoObstacle.test_forward _________________________________________________________________________________

self = <rose.server.score_test.TestNoObstacle object at 0x7fcc34e84890>

    def test_forward(self):
        for action in FORWARD_ACTIONS:
            self.player.action = action
            self.process()
>           self.assert_score(self.score)

rose/server/score_test.py:78: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <rose.server.score_test.TestNoObstacle object at 0x7fcc34e84890>, score = 0

    def assert_score(self, score):
        assert self.player.x == self.x
        assert self.player.y == self.y
>       assert self.player.score == score + config.score_move_forward
E       assert 20 == (0 + 10)
E        +  where 20 = <rose.server.player.Player object at 0x7fcc34e84310>.score
E        +    where <rose.server.player.Player object at 0x7fcc34e84310> = <rose.server.score_test.TestNoObstacle object at 0x7fcc34e84890>.player
E        +  and   10 = config.score_move_forward

rose/server/score_test.py:29: AssertionError
------------------------------------------------------------------------------------ Captured stdout call ------------------------------------------------------------------------------------
process_actions: name=A lane=0 pos=1,6 score=10 response_time=1.000000
process_actions: name=A lane=0 pos=1,6 score=20 response_time=1.000000
___________________________________________________________________________________ TestPenguin.test_other ___________________________________________________________________________________

self = <rose.server.score_test.TestPenguin object at 0x7fcc34bfd050>

    def test_other(self):
        for action in [a for a in FORWARD_ACTIONS if a != actions.PICKUP]:
            self.player.action = action
            self.process()
>           self.assert_score(self.score)

rose/server/score_test.py:118: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <rose.server.score_test.TestPenguin object at 0x7fcc34bfd050>, score = 0

    def assert_score(self, score):
        assert self.player.x == self.x
        assert self.player.y == self.y
>       assert self.player.score == score + config.score_move_forward
E       assert 20 == (0 + 10)
E        +  where 20 = <rose.server.player.Player object at 0x7fcc34bfd5d0>.score
E        +    where <rose.server.player.Player object at 0x7fcc34bfd5d0> = <rose.server.score_test.TestPenguin object at 0x7fcc34bfd050>.player
E        +  and   10 = config.score_move_forward

rose/server/score_test.py:29: AssertionError
------------------------------------------------------------------------------------ Captured stdout call ------------------------------------------------------------------------------------
process_actions: name=A lane=0 pos=1,6 score=10 response_time=1.000000
process_actions: name=A lane=0 pos=1,6 score=20 response_time=1.000000
________________________________________________________________________________ TestCrack.test_other_action _________________________________________________________________________________

self = <rose.server.score_test.TestCrack object at 0x7fcc34bfdb10>

    def test_other_action(self):
        for action in [a for a in FORWARD_ACTIONS if a != self.action]:
            self.player.action = action
            self.process()
>           self.assert_move_back()

rose/server/score_test.py:144: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <rose.server.score_test.TestCrack object at 0x7fcc34bfdb10>

    def assert_move_back(self):
        assert self.player.x == self.x
        assert self.player.y == self.y + 1
>       assert self.player.score == self.score
E       assert 10 == 0
E        +  where 10 = <rose.server.player.Player object at 0x7fcc34bfdc10>.score
E        +    where <rose.server.player.Player object at 0x7fcc34bfdc10> = <rose.server.score_test.TestCrack object at 0x7fcc34bfdb10>.player
E        +  and   0 = <rose.server.score_test.TestCrack object at 0x7fcc34bfdb10>.score

rose/server/score_test.py:44: AssertionError
------------------------------------------------------------------------------------ Captured stdout call ------------------------------------------------------------------------------------
process_actions: name=A lane=0 pos=1,7 score=0 response_time=1.000000
process_actions: name=A lane=0 pos=1,7 score=10 response_time=1.000000
________________________________________________________________________________ TestWater.test_other_action _________________________________________________________________________________

self = <rose.server.score_test.TestWater object at 0x7fcc34bcd310>

    def test_other_action(self):
        for action in [a for a in FORWARD_ACTIONS if a != self.action]:
            self.player.action = action
            self.process()
>           self.assert_move_back()

rose/server/score_test.py:144: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <rose.server.score_test.TestWater object at 0x7fcc34bcd310>

    def assert_move_back(self):
        assert self.player.x == self.x
        assert self.player.y == self.y + 1
>       assert self.player.score == self.score
E       assert 10 == 0
E        +  where 10 = <rose.server.player.Player object at 0x7fcc34bcd390>.score
E        +    where <rose.server.player.Player object at 0x7fcc34bcd390> = <rose.server.score_test.TestWater object at 0x7fcc34bcd310>.player
E        +  and   0 = <rose.server.score_test.TestWater object at 0x7fcc34bcd310>.score

rose/server/score_test.py:44: AssertionError
------------------------------------------------------------------------------------ Captured stdout call ------------------------------------------------------------------------------------
process_actions: name=A lane=0 pos=1,7 score=0 response_time=1.000000
process_actions: name=A lane=0 pos=1,7 score=10 response_time=1.000000
____________________________________________________________________________________ TestTrash.test_other ____________________________________________________________________________________

self = <rose.server.score_test.TestTrash object at 0x7fcc34bd5290>

    def test_other(self):
        for action in FORWARD_ACTIONS:
            self.player.action = action
            self.process()
            # TODO: decrease points on redundant action?
>           self.assert_move_back_no_punish()

rose/server/score_test.py:196: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <rose.server.score_test.TestTrash object at 0x7fcc34bd5290>

    def assert_move_back_no_punish(self):
        assert self.player.x == self.x
        assert self.player.y == self.y + 1
>       assert self.player.score == self.score
E       assert 10 == 0
E        +  where 10 = <rose.server.player.Player object at 0x7fcc34bd5390>.score
E        +    where <rose.server.player.Player object at 0x7fcc34bd5390> = <rose.server.score_test.TestTrash object at 0x7fcc34bd5290>.player
E        +  and   0 = <rose.server.score_test.TestTrash object at 0x7fcc34bd5290>.score

rose/server/score_test.py:49: AssertionError
------------------------------------------------------------------------------------ Captured stdout call ------------------------------------------------------------------------------------
process_actions: name=A lane=0 pos=1,7 score=0 response_time=1.000000
process_actions: name=A lane=0 pos=1,7 score=10 response_time=1.000000
____________________________________________________________________________________ TestBike.test_other _____________________________________________________________________________________

self = <rose.server.score_test.TestBike object at 0x7fcc34bdc9d0>

    def test_other(self):
        for action in FORWARD_ACTIONS:
            self.player.action = action
            self.process()
            # TODO: decrease points on redundant action?
>           self.assert_move_back_no_punish()

rose/server/score_test.py:196: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <rose.server.score_test.TestBike object at 0x7fcc34bdc9d0>

    def assert_move_back_no_punish(self):
        assert self.player.x == self.x
        assert self.player.y == self.y + 1
>       assert self.player.score == self.score
E       assert 10 == 0
E        +  where 10 = <rose.server.player.Player object at 0x7fcc34bdca50>.score
E        +    where <rose.server.player.Player object at 0x7fcc34bdca50> = <rose.server.score_test.TestBike object at 0x7fcc34bdc9d0>.player
E        +  and   0 = <rose.server.score_test.TestBike object at 0x7fcc34bdc9d0>.score

rose/server/score_test.py:49: AssertionError
------------------------------------------------------------------------------------ Captured stdout call ------------------------------------------------------------------------------------
process_actions: name=A lane=0 pos=1,7 score=0 response_time=1.000000
process_actions: name=A lane=0 pos=1,7 score=10 response_time=1.000000
___________________________________________________________________________________ TestBarrier.test_other ___________________________________________________________________________________

self = <rose.server.score_test.TestBarrier object at 0x7fcc34bdc210>

    def test_other(self):
        for action in FORWARD_ACTIONS:
            self.player.action = action
            self.process()
            # TODO: decrease points on redundant action?
>           self.assert_move_back_no_punish()

rose/server/score_test.py:196: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <rose.server.score_test.TestBarrier object at 0x7fcc34bdc210>

    def assert_move_back_no_punish(self):
        assert self.player.x == self.x
        assert self.player.y == self.y + 1
>       assert self.player.score == self.score
E       assert 10 == 0
E        +  where 10 = <rose.server.player.Player object at 0x7fcc34bdc2d0>.score
E        +    where <rose.server.player.Player object at 0x7fcc34bdc2d0> = <rose.server.score_test.TestBarrier object at 0x7fcc34bdc210>.player
E        +  and   0 = <rose.server.score_test.TestBarrier object at 0x7fcc34bdc210>.score

rose/server/score_test.py:49: AssertionError
------------------------------------------------------------------------------------ Captured stdout call ------------------------------------------------------------------------------------
process_actions: name=A lane=0 pos=1,7 score=0 response_time=1.000000
process_actions: name=A lane=0 pos=1,7 score=10 response_time=1.000000
============================================================================ 7 failed, 27 passed in 0.18 seconds =============================================================================
```

--- a/rose/server/score_test.py
+++ b/rose/server/score_test.py
@@ -2,6 +2,8 @@ from rose.common import actions, config, obstacles
 import track
 import player
 import score
+import pytest
+
 
 FORWARD_ACTIONS = [a for a in actions.ALL
                    if a not in (actions.RIGHT, actions.LEFT)]
@@ -71,12 +73,12 @@ class TestNoObstacle(SinglePlayerTest):
         self.assert_move_left()
         self.assert_keep_obstacle()
 
-    def test_forward(self):
-        for action in FORWARD_ACTIONS:
-            self.player.action = action
-            self.process()
-            self.assert_score(self.score)
-            self.assert_keep_obstacle()
+    @pytest.mark.parametrize("action", FORWARD_ACTIONS)
+    def test_forward(self, action):
+        self.player.action = action
+        self.process()
+        self.assert_score(self.score)
+        self.assert_keep_obstacle()
 
 
 class TestPenguin(SinglePlayerTest):
@@ -111,12 +113,13 @@ class TestPenguin(SinglePlayerTest):
         self.assert_keep_obstacle()
 
 
-    def test_other(self):
-        for action in [a for a in FORWARD_ACTIONS if a != actions.PICKUP]:
-            self.player.action = action
-            self.process()
-            self.assert_score(self.score)
-            self.assert_keep_obstacle()
+    @pytest.mark.parametrize(
+        "action", [a for a in FORWARD_ACTIONS if a != actions.PICKUP])
+    def test_other(self, action):
+        self.player.action = action
+        self.process()
+        self.assert_score(self.score)
+        self.assert_keep_obstacle()
 
 
 class MagicActionTest(SinglePlayerTest):
@@ -131,16 +134,14 @@ class MagicActionTest(SinglePlayerTest):
     action = None
     magic_score = None
 
-    def test_magic_action(self):
-        self.player.action = self.action
+    @pytest.mark.parametrize("action", FORWARD_ACTIONS)
+    def test_forward(self, action):
+        self.player.action = action
         self.process()
-        self.assert_score(self.score + self.magic_score)
-        self.assert_keep_obstacle()
-
-    def test_other_action(self):
-        for action in [a for a in FORWARD_ACTIONS if a != self.action]:
-            self.player.action = action
-            self.process()
+        if action == self.action:
+            self.assert_score(self.score + self.magic_score)
+            self.assert_keep_obstacle()
+        else:
             self.assert_move_back()
             self.assert_remove_obstacle()
 
@@ -188,13 +189,13 @@ class TurnTest(SinglePlayerTest):
         self.assert_move_left()
         self.assert_keep_obstacle()
 
-    def test_other(self):
-        for action in FORWARD_ACTIONS:
-            self.player.action = action
-            self.process()
-            # TODO: decrease points on redundant action?
-            self.assert_move_back_no_punish()
-            self.assert_remove_obstacle()
+    @pytest.mark.parametrize("action", FORWARD_ACTIONS)
+    def test_other(self, action):
+        self.player.action = action
+        self.process()
+        # TODO: decrease points on redundant action?
+        self.assert_move_back_no_punish()
+        self.assert_remove_obstacle()
 
 
 class TestTrash(TurnTest):

